contentgrid-spring-data-support/src/main/java/com/contentgrid/spring/data/support/auditing/v1/JwtAuditorAware.java

```diff
@@ -15,11 +15,17 @@ public Optional<UserMetadata> getCurrentAuditor() {
                 .map(SecurityContext::getAuthentication)
                 .filter(Authentication::isAuthenticated)
                 .map(Authentication::getPrincipal)
-                .filter(user -> user instanceof JwtClaimAccessor)
-                .map(JwtClaimAccessor.class::cast)
-                .filter(jwt -> jwt.getSubject() != null)
-                .filter(jwt -> jwt.getIssuer() != null)
-                .map(jwt -> new UserMetadata(jwt.getSubject(), jwt.getIssuer().toString(),
-                        jwt.hasClaim("name") ? jwt.getClaimAsString("name") : jwt.getSubject()));
+                .map(principal -> {
+                    if (principal instanceof JwtClaimAccessor jwt) {
+                        if (jwt.getSubject() != null && jwt.getIssuer() != null) {
+                            return new UserMetadata(jwt.getSubject(), jwt.getIssuer().toString(),
+                                    jwt.hasClaim("name") ? jwt.getClaimAsString("name") : jwt.getSubject());
+                        }
+                    } else if (principal instanceof String name) {
+                        // Anonymous user when 'contentgrid.security.unauthenticated.allow' is true
+                        return new UserMetadata(null, null, name);
+                    }
```

Comment:
It seems a bit dangerous to assume any principal that is a string is an unauthenticated principal?

On the other hand; if the principal is a string, we can't really get more information about the user.

The `Authentication` itself does have a `getName()` method that would return the "name" (the id/subject); which we would use in here?

Lacking more information for an anonymous user; that would mean that the subject is set to `anonymousUser`; namespace would be `null` and the display name would also be `anonymousUser`?